src/rust/lib/graph-editor/src/component/node/port/output.rs

```diff
@@ -0,0 +1,300 @@
+
+            let width  : Var<Distance<Pixels>> = shape_width.into();
+            let height : Var<Distance<Pixels>> = "input_size.y".into();
+            let width  = width  - NODE_SHAPE_PADDING.px() * 2.0;
+            let height = height - NODE_SHAPE_PADDING.px() * 2.0;
+
+            let hover_area_size   = 20.0.px();
+            let hover_area_width  = &width  + &hover_area_size * 2.0;
+            let hover_area_height = &height / 2.0 + &hover_area_size;
+            let hover_area        = Rect((&hover_area_width,&hover_area_height));
+            let hover_area        = hover_area.translate_y(-hover_area_height/2.0);
+            let hover_area        = hover_area.fill(color::Rgba::new(0.0,0.0,0.0,0.000_001));
+
+            let shrink           = 1.px() - 1.px() * &grow;
+            let radius           = 14.px();
+            let port_area_size   = 4.0.px() * &grow;
+            let port_area_width  = &width  + (&port_area_size - &shrink) * 2.0;
+            let port_area_height = &height + (&port_area_size - &shrink) * 2.0;
+            let bottom_radius    = &radius + &port_area_size;
+            let port_area        = Rect((&port_area_width,&port_area_height));
+            let port_area        = port_area.corners_radius(&bottom_radius);
+            let port_area        = port_area - BottomHalfPlane();
+            let corner_radius    = &port_area_size / 2.0;
+            let corner_offset    = &port_area_width / 2.0 - &corner_radius;
+            let corner           = Circle(&corner_radius);
+            let left_corner      = corner.translate_x(-&corner_offset);
+            let right_corner     = corner.translate_x(&corner_offset);
+            let port_area        = port_area + left_corner + right_corner;
+
+            // Move the shape so it shows the correct slice, as indicated by `offset_x`.
+            let offset_x          = Var::<Distance<Pixels>>::from(offset_x);
+            let offset_x          = width/2.0 - offset_x;
+            let port_area_aligned = port_area.translate_x(offset_x);
+
+            let overall_width     = Var::<Distance<Pixels>>::from("input_size.x");
+            let padding           = Var::<Distance<Pixels>>::from(&padding * 2.0);
+            let crop_window_width = &overall_width - &padding;
+            let crop_window       = Rect((&crop_window_width,&height));
+            let crop_window       = crop_window.translate_y(-height * 0.5);
+
+            let port_area_cropped = crop_window.intersection(port_area_aligned);
+
+            let color             = Var::<color::Rgba>::from("srgba(0.25,0.58,0.91,input_opacity)");
+            let port_area_colored = port_area_cropped.fill(color);
+
+            (port_area_colored + hover_area).into()
+        }
+    }
+}
+
+// ===========
+// === Frp ===
+// ===========
+
+
+type PortId = usize;
+
+#[derive(Clone,CloneRef,Debug)]
+pub struct Frp {
+    pub set_size        : frp::Source<Option<Vector2<f32>>>,
```

Comment:
use `V2` instead of `Vector2`. We are moving to `V2` as it has default impl defined. It is defined in math/algebra.